doc/tutorial/intro-to-autoinstall.rst

```diff
@@ -0,0 +1,243 @@
+   supported.
+
+Directly specifying autoinstall as a :code:`autoinstall.yaml` file does not
+require a :code:`#cloud-config` header, and does not use a top level
+:code:`autoinstall` key.  The autoinstall directives are placed at the top
+level.
+
+.. code-block:: yaml
+
+    version: 1
+    ....
+
+
+Cloud-init and Autoinstall interaction
+======================================
+
+While cloud-init may assist in providing the autoinstall configuration to the
+Ubuntu installer, cloud-init itself is not processing the autoinstall.
+
+If cloud-init directives are intended to modify the ephemeral system, they
+must appear at the top level of the cloud-config.  If instead
+cloud-init directives are intended to modify the system being installed, they
+must appear under a :code:`user-data` section in :code:`autoinstall`.
+
+.. code-block:: yaml
+
+    #cloud-config
+    # cloud-init directives may optionally be specified here.
+    # These directives affect the ephemeral system performing the install.
+
+    autoinstall:
+        # autoinstall directives must be specified here, not directly at the
+        # top level.  These directives are processed by the Ubuntu Installer,
+        # and configure the target system to be installed.
+
+        user-data:
+            # cloud-init directives may also be optionally be specified here.
+            # These directives also affect the target system to be installed,
+            # and are processed on first boot.
+
+
+Zero-touch deployment with autoinstall
+======================================
+
+The Ubuntu Installer contains a safeguard, intended to prevent USB Flash Drives
+with an :code:`autoinstall.yaml` file from wiping out the wrong system.
+
+During autoinstall, a prompt will be shown to confirm that the install really
+should proceed and start making modifications to the target system. ::
+
+    start: subiquity/Meta/status_GET
+    Confirmation is required to continue.
+    Add 'autoinstall' to your kernel command line to avoid this
+
+
+    Continue with autoinstall? (yes|no)
+
+To bypass this prompt, arrange for the argument :code:`autoinstall` to be
+present on the kernel command line.
+
+
+Creating an autoinstall config
+==============================
+
+When any system is installed using the Ubuntu installer, an autoinstall file
+for repeating the install is created at
+:code:`/var/log/installer/autoinstall-user-data`.
+
+
+The structure of an autoinstall configuration
+=============================================
+
+See the :ref:`ai` for full details on the supported autoinstall directives.
+
+A minimal autoinstall configuration in
+:external+cloud-init:ref:`user_data_formats-cloud_config` format looks like:
+
+.. code-block:: yaml
+
+    #cloud-config
+    autoinstall:
+        version: 1
+        identity:
+            hostname: hostname
+            username: username
+            password: $crypted_pass
+
+Here is an example file that shows off most features:
```

Comment:
I'm not clear on what "features" is referring to here

Reply:
"directives" might be a better word - it's meant to refer to the top level items shown in "Autoinstall configuration reference manual"

Reply:
I don't have very strong opinions about what word is used, as long as we define somewhere how we're using the term. So, whether we use features or directives, it should be stated up at the top that that's how we're going to refer to them (to make it completely clear what our terminology is)

Reply:
Going with "directives" for now, +1 to consistent terminology.